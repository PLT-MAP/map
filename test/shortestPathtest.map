func main(){
        Graph g = new Graph();
        Node losangeles = new Node({'temp':90,'humidity':'low'});

        Node sanjose = new Node({'temp':50,'humidity':'low'});

        Node sanfransisco = new Node({'temp':65, 'humidity':'high'});

        g.add(losangeles);

        g.add(sanjose);

        g.add(sanfransisco);

        UnDirEdge la_sj = new UnDirEdge(losangeles, sanjose, {'cost': 100});

        UnDirEdge sj_sf = new UnDirEdge(sanjose, sanfransisco, {'cost':50});

        UnDirEdge la_sf = new UnDirEdge(losangeles, sanfransisco, {'cost':180});

        g.addEdge(la_sj);

        g.addEdge(sj_sf);
        
        g.addEdge(la_sf);

        g.findShortest(losangeles, sanfransisco, 'cost');

        g.draw("shortest.jpeg");
}