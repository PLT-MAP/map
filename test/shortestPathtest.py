import networkx as nx
import sys
import matplotlib.pyplot as plt
def main():
    g = nx.MultiDiGraph()
    losangeles = 'losangeles',  {'temp':90.0,'humidity':'low'}
    sanjose = 'sanjose',  {'temp':50.0,'humidity':'low'}
    sanfransisco = 'sanfransisco',  {'temp':65.0,'humidity':'high'}
    g.add_nodes_from([(losangeles[0], losangeles[1])])
    g.add_nodes_from([(sanjose[0], sanjose[1])])
    g.add_nodes_from([(sanfransisco[0], sanfransisco[1])])
    la_sj=losangeles[0],sanjose[0], {'cost':100.0}
    sj_sf=sanjose[0],sanfransisco[0], {'cost':50.0}
    la_sf=losangeles[0],sanfransisco[0], {'cost':180.0}
    g.add_edges_from([(la_sj[0],la_sj[1],la_sj[2])])
    g.add_edges_from([(sj_sf[0],sj_sf[1],sj_sf[2])])
    g.add_edges_from([(la_sf[0],la_sf[1],la_sf[2])])
    print g.get_edge_data(losangeles[0],sanfransisco[0])
    print("Shortest path between Los Angeles and San Fransisco: ")
    try:
        print nx.shortest_path(g,losangeles[0],sanfransisco[0],'cost')
        print nx.shortest_path_length(g,losangeles[0],sanfransisco[0], 'cost')
    except:
        print 'no path'
    nx.draw(g)
    plt.show(g)
    plt.savefig("shortest.jpeg")
if __name__ == '__main__': 
	try:
		main()
	except:
		print'Error:',sys.exc_info()[0]
		print'Resolve error before running again!'