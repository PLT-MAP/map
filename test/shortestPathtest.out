import networkx as nx
import sys
import matplotlib.pyplot as plt
def main():
    g = nx.MultiDiGraph()
    losangeles = 'losangeles',  {'temp':85.0,'humidity':'low'}
    boston = 'boston',  {'temp':87.0,'humidity':'high'}
    paris = 'paris',  {'temp':80.0}
    kansascity = 'kansascity',  {'temp':40.0,'humidity':'low'}
    sanfransisco = 'sanfransisco',  {'temp':30.0,'humidity':'low'}
    durham = 'durham',  {'temp':21.0,'humidity':'low'}
    minneapolis = 'minneapolis',  {'temp':41.0}
    stpaul = 'stpaul',  {'temp':50.0,'humidity':'low'}
    philly = 'philly',  {'temp':82.0,'humidity':'low'}
    pitts = 'pitts',  {'temp':90.0}
    stpeters = 'stpeters',  {'temp':100.0}
    g.add_nodes_from([(losangeles[0], losangeles[1])])
    g.add_nodes_from([(boston[0], boston[1])])
    g.add_nodes_from([(paris[0], paris[1])])
    g.add_nodes_from([(kansascity[0], kansascity[1])])
    g.add_nodes_from([(sanfransisco[0], sanfransisco[1])])
    g.add_nodes_from([(durham[0], durham[1])])
    g.add_nodes_from([(minneapolis[0], minneapolis[1])])
    g.add_nodes_from([(stpaul[0], stpaul[1])])
    g.add_nodes_from([(philly[0], philly[1])])
    g.add_nodes_from([(pitts[0], pitts[1])])
    g.add_nodes_from([(stpeters[0], stpeters[1])])
    laboston=losangeles[0],boston[0], {'cost':550.0}
    bostonparis=boston[0],paris[0], {'cost':100.0}
    parisla=losangeles[0],paris[0], {'cost':20.0}
    g.add_edges_from([(laboston[0],laboston[1],laboston[2])])
    g.add_edges_from([(bostonparis[0],bostonparis[1],bostonparis[2])])
    g.add_edges_from([(parisla[0],parisla[1],parisla[2])])
    try:
        nx.shortest_path(g,losangeles[0],paris[0],'cost')
    except:
        print 'no path'
    print g.get_edge_data(losangeles[0],paris[0])
    if (losangeles==losangeles):
        print("hi")
    else:
        print('bye')
    print "Graph:"
    print g
    print "Nodes:"
    print g.nodes(data=True)
    print "Node losangeles:"
    print losangeles
    print "Neighbors:"
    print g.neighbors(losangeles[0])
    print "shortest paths:"
    for nodeVal in g:
        try:
            print nx.shortest_path(g, source=losangeles[0], target=nodeVal[0])
        except:
            print "no path"
        print(nx.is_isolate(g,[0]))
        empty = nx.isolates(g)
        empty_nodes = nx.MultiDiGraph()
        print(empty)
        for x in empty:
            print(x)
            empty_nodes.add_nodes_from([(x[0], x[1])])
        nx.draw(empty_nodes)
        plt.show(empty_nodes)
        plt.savefig()
if __name__ == '__main__': 
	try:
		main()
	except:
		print'Error:',sys.exc_info()[0]
		print'Resolve error before running again!'